UnityProject/Assets/Scripts/Objects/Doors/DoorMasterController.cs

```diff
@@ -661,19 +744,25 @@ public void ServerPerformInteraction(AiActivate interaction)
 
 		public bool CanOpenNetTab(GameObject playerObject, NetTabType netTabType)
 		{
-			//Only checking airlock, so when hacking UI reimplemented this check wont happen
-			//Return true so it doesnt block those checks
-			//TODO block Ai from hacking UI
-			if (netTabType != NetTabType.Airlock) return true;
+			bool isAi = playerObject.GetComponent<PlayerScript>().PlayerState == PlayerScript.PlayerStates.Ai;
+			if (netTabType == NetTabType.HackingPanel && !isAi) return true;
 
 			if (HasPower == false)
 			{
 				Chat.AddExamineMsgFromServer(playerObject, "Door is unpowered");
 				return false;
 			}
 
+			AIConnected = false;
+			HackingProcessBase.ImpulsePort(ConfirmAIConnection);
+			if (isAi && AIConnected == false)
+			{
+				Chat.AddExamineMsgFromServer(playerObject, "Door is disconnected");
+				return false;
+			}
+
 			//Only allow AI to open airlock control UI
-			return playerObject.GetComponent<PlayerScript>().PlayerState == PlayerScript.PlayerStates.Ai;
+			return isAi;
```

Comment:
```suggestion
			return true;
```

Dont need to check isAi here anymore